base/libgit2/callbacks.jl

```diff
@@ -28,16 +28,8 @@ function authenticate_ssh(creds::SSHCredentials, libgit2credptr::Ptr{Ptr{Void}},
         username_ptr, schema, host)
     isusedcreds = checkused!(creds)
 
-    errcls, errmsg = Error.last_error()
-    if errcls != Error.None
-        # Check if we used ssh-agent
-        if creds.usesshagent == "U"
-            creds.usesshagent = "E" # reported ssh-agent error, disables ssh agent use for the future
```

Comment:
This change was originally introduced in f5a097c7eb9fe354081e2a8a8b44501980457989. I'll double check that removing this doesn't break this comment:

> Don't skip using the SSH agent if there are two private packages